packages/apollo/src/utils/lists.ts

```diff
@@ -86,7 +86,19 @@ export class ListHelper {
             return undefined
         }
 
-        return existing.slice(skip, skip + first)
+        const currentList = existing.slice(skip, skip + first)
+
+        const cache = options?.cache.extract()
+
+        const hasBrokenLinks = currentList.filter(Boolean).map((link) => {
```

Comment:
should we check broken links only for the specified page (skip, first) or for full list?